libs/shared/otel/src/lib/node-tracing.ts

```diff
@@ -38,16 +48,49 @@ export class NodeTracingInitializer {
     checkServiceName(this.opts);
     checkSampleRate(this.opts);
 
-    this.provider = createNodeProvider(this.opts);
-
+    // this.provider = createNodeProvider(this.opts);
     const filter = createPiiFilter(!!this.opts?.filterPii, this.logger);
-    addGcpTraceExporter(opts, this.provider, filter, this.logger);
-    addOtlpTraceExporter(opts, this.provider, undefined, filter, this.logger);
-    addConsoleExporter(opts, this.provider, filter);
+
+    // create spanProcessors and filter undefined ones.
+    const spanProcessors = [
+      this.makeSpanProcessor(
+        getOtlpTraceExporter(this.opts, undefined, filter)
+      ),
+      this.makeSpanProcessor(getGcpTraceExporter(this.opts, filter)),
+      this.makeSpanProcessor(getConsoleTraceExporter(this.opts, filter)),
+      // add more exporters here
+    ].filter((x) => x !== undefined) as SpanProcessor[];
```

Comment:
Remove `as SpanProcessor[]`. See associated comment on `this.makeSpanProcessor`.